src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -286,12 +286,14 @@ public Pod build() {
                 	env.put("http_proxy", httpProxy);
                 }
             }
-        }
 
-        // Running on OpenShift Enterprise, security concerns force use of arbitrary user ID
-        // As a result, container is running without a home set for user, resulting into using `/` for some tools,
-        // and `?` for java build tools. So we force HOME to a safe location.
-        env.put("HOME", workingDir);
+            if (slave.getKubernetesCloud().isEnsureHomeIsValid()) {
```

Comment:
This does not seem to allow control from the `podTemplate` step.

Reply:
Drat. I shouldn't make pull requests right before going to bed 😴

Reply:
I mean, it is not a mistake, just a limitation which we may decide is good enough for now. Basically means that when defining build environments in `Jenkinsfile` rather than global settings, you do not have the opportunity to decide on a case-by-case basis whether or not to use the `$HOME` fix.